config/dependency.go

```diff
@@ -311,22 +329,18 @@ func getTerragruntOutputIfAppliedElseConfiguredDefault(dependencyConfig Dependen
 		}
 
 		if !isEmpty && dependencyConfig.shouldMergeMockOutputsWithState(terragruntOptions) {
-			stateOutputsMap, err := parseCtyValueToMap(*outputVal)
-			if err != nil {
-				return outputVal, err
-			}
-			mockOutputsMap, err := parseCtyValueToMap(*dependencyConfig.MockOutputs)
-			if err != nil {
-				return outputVal, err
+			mockMergeStrategy := dependencyConfig.getMockOutputsMergeStrategy()
+			switch mockMergeStrategy {
+			case NoMerge:
+				return outputVal, nil
+			case ShallowMerge:
+				return shallowMergeCtyMaps(*outputVal, *dependencyConfig.MockOutputs)
+			case DeepMerge:
```

Comment:
Can you introduce a new `DeepMergeMapOnly` value for the setting, as mentioned above?